Mathlib/Topology/Order/Lattice.lean

```diff
@@ -105,26 +104,270 @@ theorem Continuous.sup [Sup L] [ContinuousSup L] {f g : X → L} (hf : Continuou
+  hf.sup' hg
+
+lemma ContinuousWithinAt.sup' (hf : ContinuousWithinAt f s x) (hg : ContinuousWithinAt g s x) :
+    ContinuousWithinAt (f ⊔ g) s x :=
+  hf.sup_nhds' hg
+
+lemma ContinuousWithinAt.sup (hf : ContinuousWithinAt f s x) (hg : ContinuousWithinAt g s x) :
+    ContinuousWithinAt (fun a ↦ f a ⊔ g a) s x :=
+  hf.sup' hg
+
+lemma ContinuousOn.sup' (hf : ContinuousOn f s) (hg : ContinuousOn g s) :
+    ContinuousOn (f ⊔ g) s := fun x hx ↦
+  (hf x hx).sup' (hg x hx)
+
+lemma ContinuousOn.sup (hf : ContinuousOn f s) (hg : ContinuousOn g s) :
+    ContinuousOn (fun a ↦ f a ⊔ g a) s :=
+  hf.sup' hg
+
+end Sup
+
+section Inf
+
+variable [Inf L] [ContinuousInf L] {f g : X → L} {s : Set X} {x : X}
+
+lemma ContinuousAt.inf' (hf : ContinuousAt f x) (hg : ContinuousAt g x) :
+    ContinuousAt (f ⊓ g) x :=
+  hf.inf_nhds' hg
+
+lemma ContinuousAt.inf (hf : ContinuousAt f x) (hg : ContinuousAt g x) :
+    ContinuousAt (fun a ↦ f a ⊓ g a) x :=
+  hf.inf' hg
+
+lemma ContinuousWithinAt.inf' (hf : ContinuousWithinAt f s x) (hg : ContinuousWithinAt g s x) :
+    ContinuousWithinAt (f ⊓ g) s x :=
+  hf.inf_nhds' hg
+
+lemma ContinuousWithinAt.inf (hf : ContinuousWithinAt f s x) (hg : ContinuousWithinAt g s x) :
+    ContinuousWithinAt (fun a ↦ f a ⊓ g a) s x :=
+  hf.inf' hg
+
+lemma ContinuousOn.inf' (hf : ContinuousOn f s) (hg : ContinuousOn g s) :
+    ContinuousOn (f ⊓ g) s := fun x hx ↦
+  (hf x hx).inf' (hg x hx)
+
+lemma ContinuousOn.inf (hf : ContinuousOn f s) (hg : ContinuousOn g s) :
+    ContinuousOn (fun a ↦ f a ⊓ g a) s :=
+  hf.inf' hg
+
+end Inf
+
+section FinsetSup'
+
+variable {ι : Type*} [SemilatticeSup L] [ContinuousSup L] {s : Finset ι}
+  {f : ι → X → L} {t : Set X} {x : X}
+
+lemma ContinuousAt.finset_sup'_apply (hne : s.Nonempty) (hs : ∀ i ∈ s, ContinuousAt (f i) x) :
+    ContinuousAt (fun a ↦ s.sup' hne (f · a)) x :=
+  Tendsto.finset_sup'_nhds_apply hne hs
+
+lemma ContinuousAt.finset_sup' (hne : s.Nonempty) (hs : ∀ i ∈ s, ContinuousAt (f i) x) :
+    ContinuousAt (s.sup' hne f) x := by
+  simpa only [← Finset.sup'_apply] using finset_sup'_apply hne hs
+
+lemma ContinuousWithinAt.finset_sup'_apply (hne : s.Nonempty)
+    (hs : ∀ i ∈ s, ContinuousWithinAt (f i) t x) :
+    ContinuousWithinAt (fun a ↦ s.sup' hne (f · a)) t x :=
+  Tendsto.finset_sup'_nhds_apply hne hs
+
+lemma ContinuousWithinAt.finset_sup' (hne : s.Nonempty)
+    (hs : ∀ i ∈ s, ContinuousWithinAt (f i) t x) : ContinuousWithinAt (s.sup' hne f) t x := by
+  simpa only [← Finset.sup'_apply] using finset_sup'_apply hne hs
+
+lemma ContinuousOn.finset_sup'_apply (hne : s.Nonempty) (hs : ∀ i ∈ s, ContinuousOn (f i) t) :
+    ContinuousOn (fun a ↦ s.sup' hne (f · a)) t := fun x hx ↦
+  ContinuousWithinAt.finset_sup'_apply hne fun i hi ↦ hs i hi x hx
+
+lemma ContinuousOn.finset_sup' (hne : s.Nonempty) (hs : ∀ i ∈ s, ContinuousOn (f i) t) :
+    ContinuousOn (s.sup' hne f) t := fun x hx ↦
+  ContinuousWithinAt.finset_sup' hne fun i hi ↦ hs i hi x hx
+
```

Comment:
Aren't you missing the `Continuous` version? Same thing several times below.

Reply:
We should have a command/attribute that generates `ContinuousAt`/`ContinuousWithinAt`/`ContinuousOn` from `Tendsto` (or from `ContinuousWithinAt`).

Reply:
Fixed. Thanks!